src/wallet.c

```diff
@@ -235,24 +235,41 @@ int wallet_generate_node(const char *passphrase, const char *entropy, HDNode *no
 }
 
 
-void wallet_report_xpub(const char *keypath, char *xpub)
+int wallet_report_public(const char *keypath, char *xpub, char *pub65, char *chaincode)
```

Comment:
I guess the ideal API would be `const char *keypath, char *xpub, uint8_t pub[65], uint256 chaincode)`... then, bin2hex it at the "other side".
Otherwise we add more stuff to optimise once we move away from "everything is in hex".
But somehow xpub is already in hex and I guess we should keep it there...